.github/workflows/ts_declarations.yml

```diff
@@ -0,0 +1,67 @@
+name: TS Declarations
+
+concurrency:
+  group: wf-${{github.event.pull_request.number}}-${{github.workflow}}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  push:
+    branches: [21_2]
+
+jobs:
+  integrity-check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+    - name: Get sources
+      uses: actions/checkout@v2
+
+    - name: Use Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '15'
+
+    - name: Restore npm cache
+      uses: actions/cache@v2
+      with:
+        path: ./node_modules
+        key: ${{ runner.os }}-node-modules-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: ${{ runner.os }}-node-modules
+
+    - name: Run npm install
+      run: npm install --no-audit --no-fund
+
+    - name: Validate declarations
+      run: |
+        target=./ts/dx.all.d.ts
+        cp $target $target.current
+
+        npm run validate-declarations
+        npm run update-ts
+        
+        if ! diff $target.current $target -U 5 > $target.diff; then
+            echo "FAIL: $target is outdated:"
+            cat $target.diff | sed "1,2d"
+            exit 1
+        else
+            echo "TS is up-to-date"
+        fi
+        
+        npm run validate-ts
```

Comment:
I suggest splitting this into 2 jobs:

#### check-ts-bundle:
```yaml
        npm run update-ts
        
        if ! diff $target.current $target -U 5 > $target.diff; then
            echo "FAIL: $target is outdated:"
            cat $target.diff | sed "1,2d"
            exit 1
        else
            echo "TS is up-to-date"
        fi
```


#### validate-declarations:

```yaml
npm run validate-ts
npm run validate-declarations
```